stablehlo/tests/print_stablehlo.mlir

```diff
@@ -281,10 +281,10 @@ func.func @fft_op(%arg0: tensor<16xcomplex<f32>>) -> tensor<16xcomplex<f32>> {
 }
 
 // CHECK-LABEL: func @extensions
-func.func @extensions(%arg0 : tensor<?x?xf32, #stablehlo.type_extensions<bounds = [3, ?]>>,
+func.func @extensions(%arg0 : tensor<?x?xf32, #stablehlo.bounds<3, ?>>,
                 %arg1 : tensor<i32>) -> () {
-  // CHECK:      %0 = stablehlo.set_dimension_size %arg0, %arg1, dim = 1 : (tensor<?x?xf32, #stablehlo.type_extensions<bounds = [3, ?]>>, tensor<i32>) -> tensor<*xf32>
-  %0 = "stablehlo.set_dimension_size"(%arg0, %arg1) {dimension = 1 : i64} : (tensor<?x?xf32, #stablehlo.type_extensions<bounds = [3, ?]>>, tensor<i32>) -> tensor<*xf32>
+  // CHECK:      %0 = stablehlo.set_dimension_size %arg0, %arg1, dim = 1 : (tensor<?x?xf32, #stablehlo.bounds<3, ?>>, tensor<i32>) -> tensor<*xf32>
+  %0 = "stablehlo.set_dimension_size"(%arg0, %arg1) {dimension = 1 : i64} : (tensor<?x?xf32, #stablehlo.bounds<3, ?>>, tensor<i32>) -> tensor<*xf32>
   "stablehlo.return"() : () -> ()
```

Comment:
I tend to find bugs in similar list-printing functions when they are tested with empty input. Given that the following is a StableHLO program that passes verification, it may be worth a test that this doesn't crash or gives reasonable output (or adding a verifier check if this is invalid):

```
module {
  func.func @test(%arg0: tensor<f32, #stablehlo.type_extensions<bounds = []>>) {
    return
  }
}
```

This may fail with new syntax:

```
module {
  func.func @test(%arg0: tensor<f32, #stablehlo.bounds<>>) {
    return
  }
}
```